sdk/servicebus/service-bus/test/utils/managementUtils.ts

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {
+  QueueOptions,
+  TopicOptions,
+  SubscriptionOptions,
+  ServiceBusAtomManagementClient,
+  delay
+} from "../../src";
+
+import { EnvVarKeys, getEnvVars } from "./envVarUtils";
+
+const env = getEnvVars();
+let client: ServiceBusAtomManagementClient;
+
+/**
+ * Utility to fetch cached instance of `ServiceBusAtomManagementClient` else creates and returns
+ * a new instance constructed based on the connection string configured in environmet.
+ */
+async function getManagementClient() {
+  if (client == undefined) {
+    client = new ServiceBusAtomManagementClient(env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]);
+  }
+  return client;
+}
+
+/**
+ * Utility to apply retries to a given `operationCallBack`.
+ * Default policy is performing linear retries of upto `5` attempts that are `1000 milliseconds` apart.
+ * The retries will be preempted if given `breakConditionCallback` evaluates to `true` early on.
+ * @param operationCallback
+ * @param breakConditionCallback
+ * @param operationDescription Text describing the operation. Used for logging purposes.
+ */
+async function retry(
+  operationCallback: () => void,
+  breakConditionCallback: () => Promise<boolean>,
+  operationDescription: string
+) {
+  const retryAttempts = 5;
+  const retryDelayInMs = 1000;
+
+  let succeeded: boolean = false;
+  let count = 0;
+  while (count < retryAttempts) {
+    try {
+      const breakConditionValue = await breakConditionCallback();
+      if (breakConditionValue) {
+        succeeded = true;
+        break;
+      }
+      await operationCallback();
+    } catch (err) {
+      // Ignore error and wait before retrying
+      await delay(retryDelayInMs);
+    } finally {
+      count++;
+    }
+  }
+
+  if (!succeeded) {
+    throw new Error(`Error occurred while attempting to execute "${operationDescription}"`);
```

Comment:
At present, when we reach this line where retries have been exhausted, there is no way to determine what the error during retries was.

We should include the error that occurred during the operation in the new error that is being formed here.
I would suggest tracking the original error and prefixing error message on it with the `operationDescription`

Reply:
Updated